### DISCUSSION TITLE
Split Discord Notification Into Single Messages

### DISCUSSION THREAD

**Discussion by jimmybrancaccio** [containrrr/watchtower#1126] [Q&A]
Hello,

This is some what related to #768, in v1.1.5 I was using the Slack notification method with Discord as I believe there was no official method for sending notifications to Discord at that point. Now we have Shoutrrr which has a native Discord notification integration however it just sends everything as a single message with a ton of embeds. This unfortunately results in a bunch of lost information where the characters are being omitted:

<img width="900" alt="watchtower-shoutrrr-discord" src="https://user-images.githubusercontent.com/405585/141140484-d331304d-61f8-486a-86c6-316606942ed7.png">

I tried reverting back to using the Slack notification type (without Shoutrrr but on the latest version of Watchtower, using the Discord Webhooks URL and appending `/slack` to the end) but that didn't seem to work. Any ideas how I can set things up so each message is its own individual message without having to revert back to Watchtower v1.1.5?

Thanks!

**Comment by simonaronsson** [comment]
@piksel is probably best suited to answer this.

**Comment by piksel** [comment]
Append `?splitlines=no` to the end your notification URL, I think this will be the default in newer versions, since the multiple embeds is not really useful.
```
--notifications shoutrrr
--notification-url discord://<TOKEN>@<WEBHOOKID>?splitlines=no
```

Ref https://containrrr.dev/shoutrrr/v0.4/services/discord/#queryparam_props

**Reply by jimmybrancaccio** [reply]
Hey @piksel!

I believe I tried this but it kind of just smushed everything together. You wouldn't happen to have a screenshot of what this looks like would you? If not, no worries, I can try it again.

**Reply by piksel** [reply]
I haven't tried it with watchtower, but it looks like it doesn't use a monospace code block for the text, so perhaps it doesn't repect the newlines.

**Reply by jimmybrancaccio** [reply]
I enabled the splitlines options but it seems to just drop everything into a single embed.


<img width="854" alt="watchtower2" src="https://user-images.githubusercontent.com/405585/141651962-2c0b6548-f7c8-4b41-be52-9ac4d7a489b2.png">
<img width="1078" alt="Screen Shot 2021-11-13 at 10 39 45AM" src="https://user-images.githubusercontent.com/405585/141651937-225e05fe-4225-4b19-b31a-6d942d56366d.png">

I went back through the channel in Discord to see what it previously looked like and it looks like each message is a single message with an embed.

<img width="1099" alt="watchtower1" src="https://user-images.githubusercontent.com/405585/141651887-016da618-ce08-4b3c-bccc-b533265a5fa2.png">

Right now without using splitlines it's a single message with a ton of embeds that can result in missing data.

**Reply by piksel** [reply]
Yeah, splitlines makes a new embed for every row, and without splitlines it makes a new embed for every max characters per embed.

What it should do for splitlines is to send a new message when the maximum number of embeds are reached instead of just putting the note about omitting characters in the top.
